### PR TITLE
Added support for anlt_ll_debug

### PR DESCRIPTION
**Pull request by MichaelLilja-Xena**
AN/LT needs to have a register level based debugging interface. The class introduced, will get the base addresses from Xenaserver and utilize them to create high-level abstractions in XOA.

Signed-off-by: Michael Lilja <mli@xenanetworks.com>

### CODE REVIEW

File: xoa_driver/functions/anlt_ll_debug.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            "rx_gtm_base": ll_info.value[1],
+            "rx_serdes": ll_info.value[2],
+            "tx_gtm_base": ll_info.value[3],
+            "tx_serdes": ll_info.value[4],
+        }
+
+    async def lane_reset(self) -> None:
+        """Reset the lane (serdes)"""
+        GTM_QUAD_GT_CONFIG = 0x102
+        addr = self.ll_info.get('rx_gtm_base') + GTM_QUAD_GT_CONFIG + (self.ll_info.get('rx_serdes') * 0x40)
```
Comment (ArtemConstantinov) [inline xoa_driver/functions/anlt_ll_debug.py:47]: Can be a NoneType error, method `dict().get(<key_name>)` if not able to find an item by the provided key will return None as a default value, which can be a problem... 
Without knowing the use case example I only can recommend changing `self.ll_info` from a `dict` into a `dataclass` 

``` python
@dataclass
class LLInfo():
    base: int
    rx_gtm_base: int
    rx_serdes: int
    tx_gtm_base: int
    tx_serdes: int

...

async def init(self) -> None:
    ll_info = await commands.PL1_CFG_TMP(self.conn, self.mid, self.pid, self.lane, Layer1ConfigType.LL_DEBUG_INFO).get()
    self.ll_info = LLInfo(
        base=ll_info.value[0],
        rx_gtm_base=ll_info.value[1],
        rx_serdes=ll_info.value[2],
        tx_gtm_base=ll_info.value[3],
        tx_serdes=ll_info.value[4],
    )
    # Or an short hand but less explicit self.ll_info = LLInfo(*ll_info.value[:5])
```
then code will look like in the next example which will make the code more predictable and also use all advantages of IDE autocompletion
``` python
# addr = self.ll_info.get('rx_gtm_base') + GTM_QUAD_GT_CONFIG + (self.ll_info.get('rx_serdes') * 0x40)
addr = self.ll_info.rx_gtm_base + GTM_QUAD_GT_CONFIG + (self.ll_info.rx_serdes * 0x40)
```

File: xoa_driver/functions/anlt_ll_debug.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        return await self.__get(self.LT_TX_FRAME_REGISTER)
+
+    async def lt_tx_tf_set(self, value) -> None:
+        await self.__set(self.LT_TX_FRAME_REGISTER, value)
+
+    async def lt_rx_tf_get(self) -> int:
+        return await self.__get(self.LT_RX_FRAME_REGISTER)
+
+    async def lt_status(self) -> int:
+        return await self.__get(self.LT_RX_STATUS_REGISTER)
```
Comment (ArtemConstantinov) [inline xoa_driver/functions/anlt_ll_debug.py:95]: The repeatable code can be simplified with help of `functools.partialmethod`

```python
from functools import partialmethod

...

    mode_get = partialmethod(__get, reg=PMD_CONFIG_REGISTER)
    mode_set = partialmethod(__set, reg=PMD_CONFIG_REGISTER)

    lt_tx_config_get = partialmethod(__get, reg=LT_TX_CONFIG_REGISTER)
    lt_tx_config_set = partialmethod(__set, reg=LT_TX_CONFIG_REGISTER)

    lt_rx_config_get = partialmethod(__get, reg=LT_RX_CONFIG_REGISTER)
    lt_rx_config_set = partialmethod(__set, reg=LT_RX_CONFIG_REGISTER)

    lt_tx_tf_get = partialmethod(__get, reg=LT_TX_FRAME_REGISTER)
    lt_tx_tf_set = partialmethod(__set, reg=LT_TX_FRAME_REGISTER)

    lt_rx_tf_get = partialmethod(__get, reg=LT_RX_FRAME_REGISTER)
    lt_status = partialmethod(__set, reg=LT_RX_STATUS_REGISTER)
```

> **Notice** when we provide a class method in to `partialmethod` construction  we are not using `self`.
> For constants such as `PMD_CONFIG_REGISTER` we also can call them without self, under the particular class namespace coz there are defined as a constant which is a ClasVar, not an instance var

**Review by ArtemConstantinov — CHANGES_REQUESTED** [review]
The code is very clean and easy to understand, I pinpoint a few bugs, and the last comment is more like an tips for simplify repeatable code

**Comment by leonardhyu** [comment]
@ArtemConstantinov please review @MichaelLilja-Xena's pull request agian

**Review by ArtemConstantinov — APPROVED** [review]
> **Notice** in Driver v2.0.0 we will drop the prefix `0x` for hex values in the string format,

Means, the code as `'0x{0:08X}'.format(v)` and `'{0:02X}'.format(r)` will become:
```python
print('{0:08X}'.format(120))
print('{0:02X}'.format(10))

# also you be able to use it as

HEX_2 = '{0:02X}'.format
HEX_8 = '{0:08X}'.format
...
print(HEX_8(120))
print(HEX_2(10))
```

**Pull request merged**